.github/workflows/golangci_lint.yml

```diff
@@ -17,17 +17,22 @@ jobs:
       - name: Install golangci-lint
         run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.54.2
       
-      # go.work makes it necessary to run linter manually
       - name: Run golangci-lint
-        run: find . -name "go.mod" -execdir $(go env GOPATH)/bin/golangci-lint run --enable=gofmt --tests=false --exclude-use-default --timeout=5m0s --out-format checkstyle:golangci-lint-report.xml \;
-      
-      - name: Check golangci-lint report for errors
-        run: find . -name "golangci-lint-report.xml" -exec grep "error" {} + && exit 1 || true
+        run: docker run --rm -v ${{ github.workspace }}:/app -w /app golangci/golangci-lint:v1.54.2 golangci-lint run --max-issues-per-linter 0 --max-same-issues 0 --out-format checkstyle:golangci-lint-results.xml,json:golangci-lint-results.json || true
+
+      - name: Check golangci-lint JSON report for errors
+        run: | 
+          if jq -e '.Issues | length > 0' golangci-lint-results.json; then
+              echo "Lint issues found:" 
+              jq '.Issues[] | "\(.FromLinter): \(.Pos.Filename):\(.Pos.Line) \(.Text)"' golangci-lint-results.json
+              exit 1
+          else 
+              echo "No issues found"
+          fi
 
-      - name: Upload golangci-lint report
+      - name: Upload golangci-lint JSON report
         if: always()
         uses: actions/upload-artifact@v3
         with:
           name: golangci-lint-report
-          path: |
-            ./golangci-lint-report.xml
\ No newline at end of file
+          path: ./golangci-lint-results.xml
```

Comment:
Please, exclude this file (by RegExp) in [sonar-project.properties](https://github.com/smartcontractkit/chainlink-common/blob/main/sonar-project.properties#L6C145-L6C159), change `**/*report.xml` to `**/*results.` (note, there is no extension, so that Sonar will ignore `*results.xml` and `*reaults.json`). Otherwise, those will be scanned by Sonar.

Also, I wonder if you had a chance to check, whether golangci issues were populated in Sonar?

Reply:
Sure, I can update the sonar-project.properties. 

Yes, in the PR description I gave an example of a commit and a sonar run that resulted in golangci issues populated in sonar - are those still visible after additional workflow runs?